shell/platform/darwin/ios/framework/Headers/FlutterViewController.h

```diff
@@ -130,6 +130,15 @@ FLUTTER_EXPORT
  */
 - (id<FlutterPluginRegistry>)pluginRegistry;
 
+/**
+ * Called once the first frame has been rendered by the engine after this
+ * ViewController has appeared on the screen.
+ *
+ * This method is called before the callback set by any previous call to
+ * `setFlutterViewDidRenderCallback` is called, if any.
+ */
+- (void)firstFrameDidRender;
```

Comment:
It seems like this is just for tests.  Can you just detect the -callViewRenderedCallback message in the test instead?

Reply:
@AlbertWang0116 how do you feel about this becoming a readonly BOOL?
```
@property(nonatomic, readonly) BOOL firstFrameRendered;
```

Reply:
It works. But if the setter is ensured being called symmetrically (NO->YES->NO->YES...), EarlGrey can perform  a more optimal event-based (other than polling-based) synchronization that wait for the Flutter app to be loaded.

I think this is the case for now, and if we would like to guarantee that, probably add a doc-string to the `readwrite` version property saying that it should be assigned with NO/YES symmetrically? (So it can be the reference for future update on this class)

Reply:
@dnfield If needed, you can accomplish this KVO contract with a backing instance variable without hand-shake documentation.  Note with a getter the KVO key to observe on the unit test/Earl Grey side is `renderingFrames` and not `isRenderingFrames` (common mistake).

```
@interface FlutterViewController
@property (nonatomic, readonly, getter = isRenderingFrames) BOOL renderingFrames;
@end
```
```
@interface FlutterViewController () <FlutterBinaryMessenger>
@property (nonatomic, readwrite, getter = isRenderingFrames) BOOL renderingFrames;
@end
```
```
@implementation FlutterViewController {
...
}
@synthesize renderingFrames = _renderingFrames;
```
```
+ (BOOL)automaticallyNotifiesObserversForRenderingFrames {
    return NO;
}

- (void)setRenderingFrames:(BOOL)renderingFrames {
    if (_renderingFrames != renderingFrames) {
      [self willChangeValueForKey:@"renderingFrames"];
       _renderingFrames = renderingFrames;
      [self didChangeValueForKey:@"renderingFrames"];
    }
}
```